python/observability-online-eval/coding-agent/coding_agent/execution.py

```
@@ -0,0 +1,72 @@
+@dataclass
+class ExecutionResult:
+    success: bool
+    stdout: str
+    stderr: str
+    return_code: int
+
+
+class ExecutionEnvironment:
+    """Manages a temp directory for writing and executing generated code."""
+
+    def __init__(self, workdir: str = None):
```

Ruff (0.15.4)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)